R/zchunk_L244.building_det.R

```diff
@@ -0,0 +1,1040 @@
+
+    # Commercial floorspace
+    A44.gcam_consumer_comm <- A44.gcam_consumer %>%
+      filter(grepl("comm", A44.gcam_consumer$gcam.consumer))
+
+    L244.Floorspace_comm <- L144.flsp_bm2_R_comm_Yh %>%
+      mutate(base.building.size = round(value, energy.DIGITS_FLOORSPACE)) %>%
+      select(-value) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      mutate(gcam.consumer = A44.gcam_consumer_comm$gcam.consumer,
+             nodeInput = A44.gcam_consumer_comm$nodeInput,
+             building.node.input = A44.gcam_consumer_comm$building.node.input) %>%
+      select(region, gcam.consumer, nodeInput, building.node.input, year, base.building.size)
+
+    L244.Floorspace <- bind_rows(L244.Floorspace_resid, L244.Floorspace_comm) %>%
+      filter(year %in% BASE_YEARS)
+
+    # Demand function
+    # L244.DemandFunction_serv and L244.DemandFunction_flsp: demand function types
+    L244.DemandFunction_serv <- write_to_all_regions(A44.demandFn_serv, LEVEL2_DATA_NAMES[["DemandFunction_serv"]],
+                                                     GCAM_region_names = GCAM_region_names)
+    L244.DemandFunction_flsp <- write_to_all_regions(A44.demandFn_flsp, LEVEL2_DATA_NAMES[["DemandFunction_flsp"]],
+                                                     GCAM_region_names = GCAM_region_names)
+
+    # Floorspace demand satiation
+    # L244.Satiation_flsp: Satiation levels assumed for floorspace
+    L244.Satiation_flsp_class <- A44.satiation_flsp %>%
+      gather(sector, value, resid, comm) %>%
+      mutate(satiation.level = value * CONV_THOUS_BIL) %>%
+      select(-value)
+
+    L244.Satiation_flsp <- write_to_all_regions(A44.gcam_consumer, c("region", "gcam.consumer", "nodeInput", "building.node.input"), # replace with LEVEL2_DATA_NAMES[["BldNodes]]
+                                                GCAM_region_names = GCAM_region_names) %>%
+      # Match in the region class, and use this to then match in the satiation floorspace
+      left_join_error_no_match(A_regions %>% select(region, region.class),
+                               by = "region") %>%
+      left_join_error_no_match(L244.Satiation_flsp_class, by = c("region.class", "gcam.consumer" = "sector")) %>%
+      select(LEVEL2_DATA_NAMES[["Satiation_flsp"]])
+
+    # Satiation adder - Required for shaping the future floorspace growth trajectories in each region
+    # L244.SatiationAdder: Satiation adders in floorspace demand function
+    # First, prepare socioeconomics tables by adding region names
+    L102.pcgdp_thous90USD_Scen_R_Y <- L102.pcgdp_thous90USD_Scen_R_Y %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      rename(pcGDP_thous90USD = value)
+    L101.Pop_thous_R_Yh <- L101.Pop_thous_R_Yh %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      rename(pop_thous = value)
+
+    # In order to pass timeshift, we need floorspace for energy satiation year, which we don't have
```

Comment:
What is the energy satiation year? And, why don't we have data? Is the maximum year a reasonable default?

Reply:
`energy.SATIATION_YEAR` is just a constant. The issue is not having the data during the time shift because `energy.SATIATION_YEAR` is greater than the historical year under time shift conditions. I will clarify the comment.